File: api/src/main/java/com/mindcare/api/services/EmailService.java
package com.mindcare.api.services;

import lombok.RequiredArgsConstructor;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.stereotype.Service;
import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;

import com.mindcare.api.dto.UserRegistrationDTO;

import jakarta.mail.MessagingException;
import jakarta.mail.internet.MimeMessage;
import java.nio.charset.StandardCharsets;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@Service
@RequiredArgsConstructor
public class EmailService {

    private final JavaMailSender mailSender;
    private final TemplateEngine templateEngine;
    
    private static final String COMPANY_LOGO_URL = "http://localhost:4200/assets/logo.png";
    private static final int TOKEN_VALIDITY_HOURS = 24;

    public void sendRegistrationEmail(UserRegistrationDTO user, String token) throws MessagingException {
        Context context = new Context();
        context.setVariable("firstName", user.getFirstName());
        context.setVariable("lastName", user.getLastName());
        context.setVariable("logoUrl", COMPANY_LOGO_URL);
        context.setVariable("validationUrl", "http://localhost:4200/auth/validate?token=" + token);
        context.setVariable("expirationDate", LocalDateTime.now().plusHours(TOKEN_VALIDITY_HOURS)
            .format(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm")));

        String htmlContent = templateEngine.process("user_registration", context);
        sendHtmlEmail(user.getEmail(), "Bienvenue sur MindCare", htmlContent);
    }

    private void sendHtmlEmail(String to, String subject, String htmlContent) throws MessagingException {
        MimeMessage message = mailSender.createMimeMessage();
        MimeMessageHelper helper = new MimeMessageHelper(message, true, StandardCharsets.UTF_8.name());
        
        helper.setTo(to);
        helper.setSubject(subject);
        helper.setText(htmlContent, true);
        
        mailSender.send(message);
    }
}

File: api/app/Services/EmailService.php
<?php

namespace App\Services;

use Illuminate\Support\Facades\Mail;
use Illuminate\Support\Facades\Log;
use Illuminate\Mail\Message;
use Carbon\Carbon;

class EmailService
{
    private const COMPANY_LOGO_URL = 'http://localhost:4200/assets/logo.png';
    private const TOKEN_VALIDITY_HOURS = 24;

    public function sendRegistrationEmail(array $user, string $token): void
    {
        try {
            $expirationDate = Carbon::now()->addHours(self::TOKEN_VALIDITY_HOURS)->format('d/m/Y H:i');

            Log::info('Sending registration email', [
                'user_email' => $user['email'],
                'token' => $token,
                'validation_url' => "http://localhost:8000/api/auth/verify?token=". $token
            ]);

            Mail::send('emails.user_registration', [
                'firstName' => $user['firstname'],
                'lastName' => $user['lastname'],
                'logoUrl' => self::COMPANY_LOGO_URL,
            'validationUrl' => "http://localhost:8000/api/auth/verify?token=" . $token,
                'expirationDate' => $expirationDate
            ], function (Message $message) use ($user) {
                $message->to($user['email'])
                    ->subject('Bienvenue sur MindCare - Vérification de votre email');
            });

            Log::info('Registration email sent successfully', [
                'user_email' => $user['email']
            ]);
        } catch (\Exception $e) {
            Log::error('Failed to send registration email', [
                'error' => $e->getMessage(),
                'trace' => $e->getTraceAsString(),
                'user' => $user['email']
            ]);
            throw $e;
        }
    }

    public function sendPasswordResetEmail(array $userData, string $token)
    {
        $expirationDate = now()->addHours(24)->format('d/m/Y H:i');
        $resetUrl = env('FRONTEND_URL', 'http://localhost:4200') . '/auth/reset-password?token=' . $token;
        $logoUrl = env('APP_URL', 'http://localhost:8000') . '/images/logo.png';

        Mail::send('emails.password_reset', [
            'firstName' => $userData['firstname'],
            'lastName' => $userData['lastname'],
            'resetUrl' => $resetUrl,
            'expirationDate' => $expirationDate,
            'logoUrl' => $logoUrl
        ], function ($message) use ($userData) {
            $message->to($userData['email'])
                ->subject('Réinitialisation de votre mot de passe MindCare');
        });
    }
}
